apps/api/src/app/environments-v1/usecases/create-environment/create-environment.usecase.ts

```diff
@@ -1,24 +1,53 @@
-import { nanoid } from 'nanoid';
-import { Injectable } from '@nestjs/common';
+import { BadRequestException, Injectable, UnprocessableEntityException } from '@nestjs/common';
 import { createHash } from 'crypto';
+import { nanoid } from 'nanoid';
 
-import { EnvironmentRepository, NotificationGroupRepository } from '@novu/dal';
 import { encryptApiKey } from '@novu/application-generic';
+import { EnvironmentRepository, NotificationGroupRepository } from '@novu/dal';
 
-import { CreateEnvironmentCommand } from './create-environment.command';
-import { GenerateUniqueApiKey } from '../generate-unique-api-key/generate-unique-api-key.usecase';
+import { EnvironmentEnum, PROTECTED_ENVIRONMENTS } from '@novu/shared';
+import { CreateNovuIntegrationsCommand } from '../../../integrations/usecases/create-novu-integrations/create-novu-integrations.command';
+import { CreateNovuIntegrations } from '../../../integrations/usecases/create-novu-integrations/create-novu-integrations.usecase';
 import { CreateDefaultLayout, CreateDefaultLayoutCommand } from '../../../layouts/usecases';
+import { GenerateUniqueApiKey } from '../generate-unique-api-key/generate-unique-api-key.usecase';
+import { CreateEnvironmentCommand } from './create-environment.command';
 
 @Injectable()
 export class CreateEnvironment {
   constructor(
     private environmentRepository: EnvironmentRepository,
     private notificationGroupRepository: NotificationGroupRepository,
     private generateUniqueApiKey: GenerateUniqueApiKey,
-    private createDefaultLayoutUsecase: CreateDefaultLayout
+    private createDefaultLayoutUsecase: CreateDefaultLayout,
+    private createNovuIntegrationsUsecase: CreateNovuIntegrations
   ) {}
 
   async execute(command: CreateEnvironmentCommand) {
+    const environmentCount = await this.environmentRepository.count({
+      _organizationId: command.organizationId,
+    });
+
+    if (environmentCount >= 10) {
+      throw new BadRequestException('Organization cannot have more than 10 environments');
+    }
+
+    if (!command.system) {
```

Comment:
We don't want to make this validation when creating the default env and prod environments